### ISSUE TITLE
createGraphics throws error with WebGL

### ISSUE THREAD

**Issue by diwi** [processing/p5.js#2478]
- [x] Bug
- [x] WebGL
- [x] Desktop/Laptop
- [x] 0.5.16

The Following codesnippet contains only the really necessary parts to demonstrate a couple of issues/bugs I come across during experimenting with multiple P2D and WEBGL combinations.
Bug description is in the comment.

```javascript
//
// p5.bug, WEBGL
//
// config:
//   primary P2D
//   offscreen WEBGL
//
// throws:
//   Uncaught TypeError: Cannot read property 'apply' of undefined
//   at p5.RendererGL._applyColorBlend (p5.js:67579)
//
//
// related: 
//   https://github.com/processing/p5.js/issues/2342
//


var renderer;
var graphics;


function setup() { 

  createCanvas(800, 600, P2D);
  // createCanvas(800, 600, WEBGL);

  renderer = this._renderer;
  
  if(renderer instanceof p5.RendererGL){
    setAttributes('antialias', true);
  }

  graphics = createGraphics(800, 600, WEBGL)
} 


function draw() {
  
  // offscreen
  {
    graphics.push();
    graphics.resetMatrix();
    graphics.ortho();
    graphics.background(64);
    graphics.noStroke(); // buggy stroke 
    graphics.fill(255,0,0);
    graphics.rectMode(CENTER);
    graphics.rect(0, 0, 300, 300);
    graphics.pop();
  }
  
  
  // onscreen
  {
    push();
    background(32);
    
    if(renderer instanceof p5.Renderer2D){
      image(graphics, 0, 0);
    }
    
    if(renderer instanceof p5.RendererGL){
      resetMatrix();
      ortho();
      texture(graphics);
      plane(width, height);
      // 0/0 ... top left corner
      translate(-width/2,-height/2);
    }
    

    // rect attached to mouse
    translate(mouseX, mouseY);
    
    // tilt to see Aliasing
    if(renderer instanceof p5.Renderer2D) rotate (3 * PI/180);
    if(renderer instanceof p5.RendererGL) rotateZ(3 * PI/180);
    
    rectMode(CENTER);
    noStroke();
    fill(0,255,0);
    rect(0, 0, 50, 300);
    pop();
  }
  
}




p5.RendererGL.prototype._applyColorBlend = function(v1,v2,v3,a){
  var gl = this.GL;
  
  // < fix, ... not really, but at least i got the demo going with this
  var inst = this._pInst;
  if(inst instanceof p5.Graphics){
    var inst = inst._pInst;
  }
  // fix >
  

  var color = inst.color.apply(inst, arguments);
  var colors = color._array;
  if(colors[colors.length-1] < 1.0){
    gl.depthMask(false);
    gl.enable(gl.BLEND);
    gl.blendEquation( gl.FUNC_ADD );
    gl.blendFunc( gl.SRC_ALPHA, gl.ONE_MINUS_SRC_ALPHA );
  } else {
    gl.depthMask(true);
    gl.disable(gl.BLEND);
  }
  return colors;
};


```

**Comment by Spongman** [comment]
two things are happening here:
1) the `p5.Graphics` is creating its renderer (in its constructor) before it has monkey-patched its prototype to look like a `p5` instance.
2) the GL renderer is trying to create colors, and the color parser is looking for the renderer's colorMode, but there isn't a renderer yet => boom.

#1 is simple to fix. #2 can be worked around, but i think a bigger question here is: does it make sense to store instance state (like colorMode, ellipseMode, etc...) in the renderer? or should these be stored in the instance so it's not lost when switching renderers?

**Comment by mlarghydracept** [comment]
renamed for clarity

**Comment by mlarghydracept** [comment]
It seems like there are a few [2D renderer variables stored in the p5 instance ](https://github.com/processing/p5.js/blob/master/src/core/core.js#L183) but am not sure the conversation that led to that. But it might be better to change the order that p5.Graphics is creating its renderer in? @lmccart thoughts?

**Comment by Spongman** [comment]
i moved those detail variables in there since previously they were global variables (which are problematic when multiple instances/graphics exist). i figured they would be better in the instance than in the renderer since presumable 3d beziers will have the same detail settings eventually.

**Comment by Spongman** [comment]
ok, i fixed this in #2502.

in order to able to create colors without a renderer, i moved the _colorMode & _colorMaxes properties out of the renderer and into the instance (& graphics).

i can see that this could quite easily be bad for some reason. however, it might be worth considering whether or not all those style properties (the common ones pushed in `push()`) could live in the instance instead of the renderer.
